### PR TITLE
Include explicit attrs when default_attrs=true

### PR DESCRIPTION
**Pull request by smangelsdorf**
With the `default_attrs` option set to `true`, xmerl_scan was replacing the attributes for each element with the default attributes, discarding any attributes which were explicitly set.

The new behaviour appends the default attributes, keeping the explicit attributes intact.

The new test I've added for this change tests two documents to ensure both attributes exist in each case:

```xml
<!DOCTYPE doc [<!ATTLIST doc b CDATA "default">]>
<doc a="explicit"/>
```

```xml
<!DOCTYPE doc [<!ATTLIST doc b CDATA "default">]>
<doc b="also explicit" a="explicit"/>
```

Prior to this change, the `a` attribute was not included in the parsed structure.

### CODE REVIEW

File: lib/xmerl/src/xmerl_scan.erl [hunk 1]
```diff
@@ -2225,16 +2225,18 @@ processed_whole_element(S=#xmerl_scanner{hook_fun = _Hook,
...
+                                parents = [{Name, Pos} | Parents],
+                                language = Lang,
+                                nsinfo = NSI,
+                                namespace = Namespace,
+                                value = AttValue,
+                                normalized = true} ||
+                  {AttName, AttValue} <- get_default_attrs(S, Name),
+                  AttValue =/= no_value,
+                  not lists:keymember(AttName, #xmlAttribute.name, Attrs) ],
+            lists:append(Attrs, DefaultAttrs);
```
Comment (fenollp) [inline lib/xmerl/src/xmerl_scan.erl]: If you do `Attrs ++ DefaultAttrs` the compiler will optimize the list comprehension AFAIK.

Comment (smangelsdorf) [inline lib/xmerl/src/xmerl_scan.erl]: Thanks for the tip, I had no idea there would be a difference (my Erlang skills are still a bit green). I'd be happy to address this in another PR since this has already been merged, but I'd love to convince myself first that it was actually an improvement. Any tips on how I could observe the difference?
